eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -198,6 +206,22 @@ jobs:
         displayName: 'product build'
 
 
+    - ${{ if eq(parameters.runtimeFlavor, 'mono') }}:
+      # We need to explictly download CoreCLR for Mono
+      - template: /eng/pipelines/common/download-artifact-step.yml
+        parameters:
+          unpackFolder: $(coreClrProductRootFolderPath)
+          artifactFileName: '$(coreClrProductArtifactName)$(archiveExtension)'
+          artifactName: '$(coreClrProductArtifactName)'
+          displayName: 'CoreCLR product download for Mono'
+
+      - script: $(_msbuildCommand)
+                $(Build.SourcesDirectory)/src/mono/mono.proj
+                /t:PatchCoreClrCoreRoot
+                /p:Configuration=$(buildConfig);BuildArchitecture=$(archType)
```

Comment:
We should be careful with passing arguments separated by `;` as if we want to enable this in Windows I would suspect PS failing to parse this and will  think they're separate args to the build script. Let's use two `/p:` statements instead.

Reply:
Also, `BuildArchitecture` usually relates to the architecture the machine we're building on. We should use `TargetArchitecture` instead.